bgzf/index/index.go

```diff
@@ -58,24 +62,34 @@ func (r *ChunkReader) Read(p []byte) (int, error) {
 	// the current chunk. We do not need to consider reading
 	// beyond the end of the block because the bgzf.Reader is in
 	// blocked mode and so will stop there anyway.
-	if r.r.LastChunk().End.File == r.chunks[0].End.File {
-		p = p[:min(len(p), int(r.chunks[0].End.Block-r.r.LastChunk().End.Block))]
+	// To do this we set the cursor to the end of the last read
+	// if we are still in the same block, and truncate p to the
+	// length of the remaining chunk data.
+	var cursor uint16
+	if last.Begin.File == r.chunks[0].Begin.File {
+		cursor = last.End.Block
```

Comment:
If I change this to:
              cursor = r.chunks[0].Begin.Block

Then my tests (doing tons of intersections and comparing to tabix CLI tool) are getting a lot farther, but still failing eventually. I think it's in the block below for switching chunks to early??